src/test/java/ch/wisv/areafiftylan/TicketTransferRestIntegrationTest.java

```diff
@@ -0,0 +1,335 @@
+import ch.wisv.areafiftylan.model.util.TicketType;
+import ch.wisv.areafiftylan.security.TicketTransferToken;
+import ch.wisv.areafiftylan.service.repository.TicketRepository;
+import ch.wisv.areafiftylan.service.repository.TicketTransferTokenRepository;
+import ch.wisv.areafiftylan.util.SessionData;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.response.Response;
+import org.junit.Assert;
+import org.apache.http.HttpStatus;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.keygen.StringKeyGenerator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.jayway.restassured.RestAssured.given;
+
+/**
+ * Created by beer on 5-1-16.
+ */
+public class TicketTransferRestIntegrationTest extends IntegrationTest{
+    private final String TRANSFER_ENDPOINT = "/tickets/transfer";
+
+    private User outsider;
+    private User ticketReceiver;
+    private Ticket ticket;
+
+    @Autowired
+    private TicketRepository ticketRepository;
+
+    @Autowired
+    private TicketTransferTokenRepository tttRepository;
+
+
+    @Before
+    public void initTransferTest(){
+        outsider = makeOutsider();
+        ticketReceiver = makeTicketReceiver();
+        ticket = makeTicket();
+    }
+
+    @After
+    public void cleanTransferTest(){
+        tttRepository.deleteAll();
+        ticketRepository.deleteAll();
+    }
+
+    private User makeTicketReceiver(){
+        User receiver = new User("receiver", new BCryptPasswordEncoder().encode("password"), "receiver@gmail.com");
+        receiver.getProfile()
+                .setAllFields("receiver", " of tickets", "GotYaTicket", Gender.MALE, "Money Owner 4", "2826GJ", "Tomorrowland",
+                        "0906-1111", null);
+
+        return userRepository.saveAndFlush(receiver);
+    }
+
+    private User makeOutsider(){
+        User outsider = new User("outsider", new BCryptPasswordEncoder().encode("password"), "outsider@gmail.com");
+        outsider.getProfile()
+                .setAllFields("Nottin", "Todoeo Witit", "Lookinin", Gender.FEMALE, "LoserStreet 1", "2826GJ", "China",
+                        "0906-3928", null);
+
+        return userRepository.saveAndFlush(outsider);
+    }
+
+    private Ticket makeTicket(){
+        Ticket t = new Ticket(user, TicketType.EARLY_FULL, false, false);
+        t = ticketRepository.saveAndFlush(t);
+
+        return t;
+    }
+
+    @Test
+    public void testAddTransfer_Anon(){
+        Map<String, String> transferRequest = new HashMap<>();
+        transferRequest.put("goalUsername", ticketReceiver.getUsername());
+
+        given().
+        when().
+                content(transferRequest + "/" + ticket.getId()).contentType(ContentType.JSON).
```

Comment:
This is weird. Test is wrong, but the outcome is correct anyway because there's not CSRF token

```
{
    "timestamp": 1457724563201,
    "status": 403,
    "error": "Forbidden",
    "message": "Expected CSRF token not found. Has your session expired?",
    "path": "/tickets/transfer"
}
```